Lib/tarfile.py

```diff
@@ -513,21 +513,10 @@ def seek(self, pos=0):
             raise StreamError("seeking backwards is not allowed")
         return self.pos
 
-    def read(self, size=None):
-        """Return the next size number of bytes from the stream.
-           If size is not defined, return all bytes of the stream
-           up to EOF.
-        """
-        if size is None:
```

Comment:
I have only one question. Why this branch was removed?

Reply:
This issue is follow up of bpo-34010, (GH-8020).
See also, https://bugs.python.org/issue34010#msg321040

Reply:
Thank you. Instances of `_Stream` are never leaked to the end user? Then this change LGTM.

Reply:
I saw that the class is private, but I didn't notice that TarFile.fileobj is public. Maybe just replace the assertion with a regular if/raise, just in case? Maybe raise a NotImplementedError?